core/nodejsActionBase/src/service.js

```diff
@@ -112,10 +113,14 @@ function NodeActionService(config) {
      */
     this.runCode = function runCode(req) {
         if (status === Status.ready) {
-            setStatus(Status.running);
+            if (!ignoreRunStatus) {
+                setStatus(Status.running);
```

Comment:
small nit --- how about just moving the predicate into `setStatus` itself 

```js
function setStatus(newStatus) {
  if (status !== Status.stopped && !ignoreRunStatus) {
    status = newStatus;
  }
}
```

Reply:
I don't think that we want to ignore status in all cases, only when we are setting/testing `Status.running`. This seemed less confusing, with that in mind - WDYT?